src/game/requester.c

```diff
@@ -6,27 +6,38 @@
 #include "global/const.h"
 #include "global/types.h"
 
+#include <libtrx/memory.h>
+
+#include <stdarg.h>
 #include <stdbool.h>
+#include <stdio.h>
 #include <string.h>
 
 #define BOX_BORDER 2
 #define BOX_PADDING 10
 
-void Requester_Init(REQUEST_INFO *req)
+void Requester_Init(REQUEST_INFO *req, const uint16_t max_items)
 {
-    req->heading = NULL;
-    req->background = NULL;
-    req->moreup = NULL;
-    req->moredown = NULL;
-    for (int i = 0; i < MAX_REQLINES; i++) {
-        req->texts[i] = NULL;
-        req->item_flags[i] = 0;
-    }
+    req->max_items = max_items;
+    req->heading_text = Memory_Alloc(sizeof(char) * req->item_text_len);
+    req->item_blocked = Memory_Alloc(sizeof(bool) * max_items);
+    req->item_texts = Memory_Alloc(sizeof(char *) * max_items);
+    Requester_ClearTextstrings(req);
```

Comment:
Should I need to traverse each `item_blocked` and `item_texts` and initialize them to `NULL`? Should I also be doing that in `Requester_ClearTextstrings`?

Reply:
No need to initialize to NULL, since `Memory_Alloc` zeroes the memory for us.
I don't think `ClearTextstrings` is required to deallocate the memory, it's only important that we don't leave it dangling and call the `Shutdown` route.